third-party/fftw3/CMakeLists.txt

```
@@ -0,0 +1,69 @@
+therock_subproject_fetch(therock-fftw3-sources
+  CMAKE_PROJECT
+  # Originally mirrored from: https://fftw.org/fftw-3.3.10.tar.gz
+  URL https://rocm-third-party-deps.s3.us-east-2.amazonaws.com/fftw-3.3.10.tar.gz
+  URL_HASH SHA256=56c932549852cddcfafdab3820b0200c7742675be92179e59e6215b340e26467
+)
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+  set(FFTW3_ENABLE_THREADS TRUE CACHE BOOL "FFTW3_ENABLE_THREADS set TRUE on Linux" FORCE)
+else()
+  set(FFTW3_ENABLE_THREADS FALSE CACHE BOOL "FFTW3_ENABLE_THREADS set FALSE" FORCE)
+endif()
```

```suggestion
if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
  set(_fftw_enable_threads "ON")
else()
  set(_fftw_enable_threads "OFF")
endif()
```
I don't see a strict need to make this a cached variable. Let's rather follow the style at other places and make it a local variable which is passed. Furthermore following https://github.com/FFTW/fftw3/blob/adde9bec41206a3d6bbb02bdbf9f64726d7d2009/CMakeLists.txt#L17, let's use `ON` and `OFF` here instead.